hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordMerger.java

```diff
@@ -167,17 +172,16 @@ default String[] getMandatoryFieldsForMerging(HoodieTableConfig cfg) {
    */
   String getMergingStrategy();
 
-  /**
-   * The record merge mode that corresponds to this record merger
-   */
-  default RecordMergeMode getRecordMergeMode() {
-    switch (getMergingStrategy()) {
-      case DEFAULT_MERGER_STRATEGY_UUID:
-        return RecordMergeMode.EVENT_TIME_ORDERING;
-      case OVERWRITE_MERGER_STRATEGY_UUID:
-        return RecordMergeMode.OVERWRITE_WITH_LATEST;
+  static String getAvroMergerStrategyFromMergeMode(RecordMergeMode mergeMode) {
+    switch (mergeMode) {
+      case OVERWRITE_WITH_LATEST:
+        return OVERWRITE_MERGER_STRATEGY_UUID;
+      case EVENT_TIME_ORDERING:
+        return DEFAULT_MERGER_STRATEGY_UUID;
+      case CUSTOM:
+        return PAYLOAD_BASED_MERGER_STRATEGY_UUID;
```

Comment:
Should all modes go through `PAYLOAD_BASED_MERGER_STRATEGY_UUID` since `OVERWRITE_WITH_LATEST` should use `OverwriteWithLatestAvroPayload` and `EVENT_TIME_ORDERING` should use `DefaultHoodieRecordPayload` if `HoodieAvroRecordMerger` is used?

Reply:
Note to myself for revisiting.